internals/src/const_casts.rs

```diff
@@ -0,0 +1,77 @@
+//! Const-compatible integer casting functions.
+//!
+//! This module provides explicit, const-compatible functions for integer type conversions
+//! that would normally be done using the [`Into`] trait. Since trait methods cannot be used
+//! in `const` contexts, these functions serve as alternatives that make conversion intent
+//! clear while maintaining compile-time evaluation capabilities.
+//!
+//! All functions in this module perform the same conversions as their corresponding
+//! [`From`] implementations. Platform-specific functions are feature-gated to prevent
+//! compilation on unsupported architectures.
```

Comment:
This is not correct fro multiple reasons.

Reply:
Thank you for your feedback, very valuable. I agree it I was confused and I didn't read the issue right, but I was confused by this statement: "This issue is easy in principle but involves a bunch of changes." - I was thinking that many changes are expected, so went through all the casts in the code to make them all use the same casting module. I hope the next commit will be correct.